### PR TITLE
Update renovate.json

### PR DESCRIPTION
**Pull request by lmilbaum**
**What this PR does / why we need it**:

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

**Jira Ticket**:

```jira-ticket

```

**Release note**:

```release-note
Renovate configuration - Do not pin centos:stream9 digest because this image is not renovate friendly and disable renovating gomod manager
```

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -37,6 +37,13 @@
        "matchManagers": [ "pip_requirements" ],
        "groupName": "pip_requirements"
     },
+    {
+      "description": "Group go updates",
+      "matchManagers": ["gomod"],
+      "matchDepTypes": ["major", "minor", "patch", "digest", "indirect"],
+      "groupName": "go updates",
+      "postUpdateOptions": ["gomodUpdateImportPaths", "gomodTidy"]
+    },
```
Comment (nunnatsa) [inline renovate.json:46]: What does this change do? This is not part of the PR description.

Comment (lmilbaum) [reply renovate.json:46]: Sorry. It shouldn't be here. Removing

**Comment by nunnatsa** [comment]
@lmilbaum - there are still several changes that are not documented in the PR description and it's not clear what this PR changes.

**Review by nunnatsa — APPROVED** [review]
